miniapp/retrieve_kernels.cxx

```diff
@@ -123,6 +123,50 @@ namespace {
    } // end of retrieve_data_cuda
 
 #endif
+
+#if defined(RAJA_ENABLE_HIP)
+   void retrieve_data_hip(const int nqpts, const int nstatev,
+                          const double* stress_svec_p_array, const double* vol_ratio_array,
+                          const double* eng_int_array, double* state_vars_array,
+                          double* stress_array){
+      const int ind_int_eng = nstatev - ecmech::ne;
+      const int ind_vols = ind_int_eng - 1;
+
+      RAJA::RangeSegment default_range(0, nqpts);
+
+      RAJA::forall<RAJA::hip_exec<384> >(default_range, [ = ] RAJA_DEVICE(int i_qpts) {
```

Comment:
This is a lot of code duplication for what appears to be a on-line difference in the RAJA::forall invocation. 

Is there some not-overly-painful way to template this to avoid the code duplication?

Reply:
I do know how to eliminate all of the code duplication using templates as we've got the code for it in SNLS.

I think for the time being it'll be okay to keep it as is just because I'm looking to do a series of refactors on ExaCMech to reduce a lot of boilerplate code and make code maintenance hopefully simpler. As part of those refactors, one set of changes will be to require the raja portability suite, so we can leverage all the new abstractions in SNLS and some of the new ones being added as part of https://github.com/LLNL/SNLS/pull/14. With that refactor all of these forall's we have in the code will be reduced down to a single `snls::forall` which handles all the backends for us.

Reply:
Actually ended up reducing a lot of these GPU calls down to only a few new lines and removed a ton of the repeated lines of code.